File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.py
class Solution(object):
    def kthDistinct(self, arr, k):
        """
        :type arr: List[str]
        :type k: int
        :rtype: str
        """
        l=[]
        for i in range(len(arr)):
            if arr.count(arr[i])==1:
                l.append(arr[i])
            else:
                pass
        if len(l)>=k:
            return l[k-1]
        return ""
        

File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.java
import java.util.*;

class Solution {
    public String kthDistinct(String[] arr, int k) {
        Map<String, Integer> count = new HashMap<>();
        for (String s : arr) {
            count.put(s, count.getOrDefault(s, 0) + 1);
        }
        
        int distinctCount = 0;
        for (String s : arr) {
            if (count.get(s) == 1) {
                distinctCount++;
                if (distinctCount == k) {
                    return s;
                }
            }
        }
        
        return "";
    }
}